040_d3_L54/040_d3_L54_safe.py

```
@@ -0,0 +1,154 @@
+        # Sanitize error messages for logging to prevent log injection
+        safe_http_err = re.sub(r'[\r\n\t]', '_', str(http_err))
+        logging.error(f'HTTP error occurred: {safe_http_err}')
+        logging.debug(f'Full HTTPError exception: {repr(http_err)}')
+    except Exception as err:
+        # Sanitize error messages for logging to prevent log injection
+        safe_err = re.sub(r'[\r\n\t]', '_', str(err))
+        logging.error(f'An error occurred: {safe_err}')
+        logging.debug(f'Full exception: {repr(err)}')
+
+
+def upload_memory(text: str, device_name: str = "Discord Server API Connection", created_time: Optional[str] = None):
```

Ruff (0.14.2)

Unused function argument: `device_name`

(ARG001)